src/emqtt_bench.erl

```diff
@@ -592,8 +592,10 @@ maybe_retry(Parent, N, PubSub, Opts, ContinueFn) ->
     end.
 
 loop(Parent, N, Client, PubSub, Opts) ->
-    Idle = max(proplists:get_value(interval_of_msg, Opts, 0) * 2, 500),
+    Interval = proplists:get_value(interval_of_msg, Opts, 0),
+    Idle = max(Interval * 2, 500),
     MRef = proplists:get_value(publish_signal_mref, Opts),
+    ok = put_publish_begin_time(),
```

Comment:
I think this should be placed in `publish/2` as before.  This point is not necessarily when something will be published.

In particular, if one uses the option to only start publishing [after all publishers are spawned](https://github.com/emqx/emqtt-bench/pull/182/files#diff-3e65fcfb5555d19578c352508532dabd1b40319895276ce57c8335c9e7fc2b7fR600), this time will not be accurate.

Reply:
Good point